kafka/integration_test.go

```diff
@@ -1208,6 +1211,38 @@ func TestProducerConsumerHeaders(t *testing.T) {
 	}
 
 	c.Close()
+
+}
+
+type AdminTestSuite struct {
+	suite.Suite
+	compose *testcontainers.LocalDockerCompose
+}
+
+func (ats *AdminTestSuite) SetupSuite() {
+	testconfInit()
+	if !testconf.Docker {
+		return
+	}
+	ats.compose = testcontainers.NewLocalDockerCompose([]string{"testresources/docker-compose.yaml"}, "test-docker")
```

Comment:
the two other occasions i've needed to deal with the test code being responsible for managing the lifecycle of the docker containers it makes use of, it's brought more of a pain than a benefit.

that makes me wary of doing this.

it's not very hard to require the user manually do this, or make the script running the tests responsible for it.

Reply:
Based on the time it took me to get this to a working state, I'm inclined to agree, it's probably a lot of effort to get it running, especially on the CI. We should just maybe leave the docker-compose file and make the rest of the work be done by a script.